awx/main/analytics/core.py

```diff
@@ -140,3 +140,6 @@ def ship(path):
         logger.exception('{} failure:'.format(cmd))
     except subprocess.TimeoutExpired:
         logger.exception('{} timeout:'.format(cmd))
+    finally:
```

Comment:
What about the `if shutil.which(agent) is None:` case above?  Might be better to wrap the *entire* function in one big `try:` block with an `except:` at the end:

```python
def ship(path): 
    try:
        agent = '...'
        ....
        try:
            cmd = ...
        except subprocess.CalledProcessError:
            ....
        except ...
            ....
    finally:
        os.remove() 
```